### PR TITLE
Add Custom Fields support in Products Yosemite layer

### LINKED ISSUE #13833: Add MetaData to Product in Yosemite

### PR DESCRIPTION
**Pull request by pmusolino**
Closes: #13833

## Description
This pull request adds support for custom fields in the products Yosemite layer. It enhances the existing product model to include custom fields and updates various layers to handle the new data appropriately. This also merges the encoding of metadata (`customFields` + `subscription`).

## Steps to reproduce
1. Go on the web, and create a product with subscription with also some custom fields.
2. Open the product you created on mobile.
3. Try to edit the information under the subscription, and save.
4. Make sure the subscription changes are applied correctly.
5. Go on the web, and make sure the custom fields are still there.
6. Remove one custom field from the web, then refresh the app.
7. The custom field deleted should have disappeared from the specific product.

## Testing information
I conducted testing by first creating a product with a subscription and custom fields on the web. I then opened the product on iOS simulator and edited the subscription information, ensuring the changes were saved correctly. Finally, I verified on the web that the custom fields remained intact and tried to delete a custom field on the web, to make sure changes are reflected on mobile. The testing covered both web and mobile platforms to ensure consistency and data integrity across different devices.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by hafizrahman** [comment]
Testing notes:

Tested using a WordPress.com site, I followed the **Steps to Reproduce** part and everything works correctly.

File: Networking/Networking/Model/Product/Product.swift [hunk 1]
```diff
@@ -117,7 +117,7 @@ public struct Product: Codable, GeneratedCopiable, Equatable, GeneratedFakeable
     // MARK: Subscription Product properties
 
     /// Subscription settings. Applicable to subscription-type products only.
-    public let subscription: ProductSubscription?
+    public let subscription: ProductSubscription? //TODO: migrate AddOns to MetaData
```
Comment (hafizrahman) [inline Networking/Networking/Model/Product/Product.swift]: np: should say Subscriptions instead of AddOns?

**Review by hafizrahman — APPROVED** [review]
Code looks good to me, just left a small np.

File: Networking/Networking/Model/Product/Product.swift (hunk 1 above)
Comment (pmusolino) [reply Networking/Networking/Model/Product/Product.swift]: Yes, good catch. Edited here 61bd5ab44e72071fd0d0545831e2c0f7121251db

**Pull request merged**